### ISSUE TITLE
 Module._resolveLookupPaths is not a function | dd-trace

### ISSUE THREAD

**Issue by diavrank** [oven-sh/bun#8839]
### What version of Bun is running?

1.0.26+3e1231deb

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

1. Create a Nestjs app
2. Install dd-trace 2.15.0
3. Run app with: bun start:dev
```ts
"start:dev": "NODE_ENV=localdev nest start --watch --exec \"bun run\"",
```
4. See the error:

````shell
[12:01:25 PM] Starting compilation in watch mode...

[12:01:44 PM] Found 0 errors. Watching for file changes.

110 | 
111 |       hooks = moduleHooks[name]
112 |       if (!hooks) return exports // abort if module name isn't on whitelist
113 | 
114 |       // figure out if this is the main module file, or a file inside the module
115 |       const paths = Module._resolveLookupPaths(name, this, true)
                          ^
TypeError: Module._resolveLookupPaths is not a function. (In 'Module._resolveLookupPaths(name, this, !0)', 'Module._resolveLookupPaths' is undefined)
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:115:21
      at /Users/diavrank/app/node_modules/@aws-sdk/client-acm-pca/node_modules/@smithy/smithy-client/dist-cjs/index.js:4:9
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:85:21
      at /Users/diavrank/app/node_modules/@aws-sdk/client-acm-pca/dist-cjs/ACMPCAClient.js:13:7
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:85:21
      at /Users/diavrank/app/node_modules/@aws-sdk/client-acm-pca/dist-cjs/index.js:5:9
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:85:21
````

### What is the expected behavior?

Nest app server should start correctly on port 3000.

### What do you see instead?

````shell
[12:01:25 PM] Starting compilation in watch mode...

[12:01:44 PM] Found 0 errors. Watching for file changes.

110 | 
111 |       hooks = moduleHooks[name]
112 |       if (!hooks) return exports // abort if module name isn't on whitelist
113 | 
114 |       // figure out if this is the main module file, or a file inside the module
115 |       const paths = Module._resolveLookupPaths(name, this, true)
                          ^
TypeError: Module._resolveLookupPaths is not a function. (In 'Module._resolveLookupPaths(name, this, !0)', 'Module._resolveLookupPaths' is undefined)
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:115:21
      at /Users/diavrank/app/node_modules/@aws-sdk/client-acm-pca/node_modules/@smithy/smithy-client/dist-cjs/index.js:4:9
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:85:21
      at /Users/diavrank/app/node_modules/@aws-sdk/client-acm-pca/dist-cjs/ACMPCAClient.js:13:7
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:85:21
      at /Users/diavrank/app/node_modules/@aws-sdk/client-acm-pca/dist-cjs/index.js:5:9
      at /Users/diavrank/app/node_modules/dd-trace/packages/dd-trace/src/ritm.js:85:21
````

### Additional information

I was getting this another [error](https://github.com/oven-sh/bun/issues/6374) previously so now I'm getting this new error.

I have installed Node 18.16.1 in my machine as well but I am using Bun to install the deps and run the app as I mention above.

**Comment by jttommy** [comment]
still an issue as of v1.1.4

**Comment by Jarred-Sumner** [comment]
Fixed by @paperdave in #10568
